garrysmod/gamemodes/terrortown/gamemode/init.lua

```diff
@@ -875,7 +881,7 @@ function SelectRoles()
       -- make this guy traitor if he was not a traitor last time, or if he makes
       -- a roll
       if IsValid(pply) and
-         ((prev_roles[pply] ~= ROLE_TRAITOR) or (math.random(1, 3) == 2)) then
+         ((not table.HasValue(prev_roles[ROLE_TRAITOR], pply)) or (math.random(1, 3) == 2)) then

```

Comment:
use hashtables

Reply:
why?

Reply:
Because hashtables are faster then iterating thru the whole table.

replace `table.insert(prev_roles[r], v)` by `prev_roles[r][v] = true`

Reply:
And obviously, do the correct changes in the code

Reply:
His improvements did use hashtables. You're referencing his revert commit.

Reply:
no u

Reply:
> His improvements did use hashtables. You're referencing his revert commit.

Yes, you're right, I returned the commit because I could not create another branch.

Reply:
> no u

Yes, I returned the commit because I could not create another branch.